### PR TITLE
Silence errors on overwriting GeoJSON

### LINKED ISSUE #771: Overwriting GeoJSON logs an ERROR but succeeds
## Expected behavior and actual behavior.

Overwriting a GeoJSON file results in a `logging` error message about the file not existing.

## Steps to reproduce the problem.

Same example code as in https://github.com/Toblerity/Fiona/issues/438#issuecomment-343154689 but with logging enabled.

```
import logging

logging.basicConfig(
    level=logging.INFO,
)

import fiona
from fiona.crs import from_epsg
from shapely.geometry import Point, mapping

schema = {
    "geometry": "Point",
    "properties": {"id": "int"}
}

feature = {
    "geometry": mapping(Point(0,0)),
    "properties": {"id": "0"}
}

for i in range(2):
    with fiona.open("/tmp/test.geojson", "w", 
                    driver="GeoJSON",
                    schema=schema, 
                    crs=from_epsg(4326)) as sink:
        sink.write(feature)
        print(f"Write #{i+1} succeeded")
```

results in

```
Write #1 succeeded
ERROR:fiona._env:/tmp/test.geojson: No such file or directory
WARNING:fiona._env:driver GeoJSON does not support creation option ENCODING
Write #2 succeeded
```

The error message is false. The file exists after the first successful write. At least from my user perspective.

## Operating system

Archlinux

## Fiona and GDAL version and provenance

gdal 3.0.0-2 from Archlinux repositories.
Fiona 1.8.6 via pip.

### PR DESCRIPTION
**Pull request by sgillies**
Resolves #771

### CODE REVIEW

File: fiona/ogrext.pyx [hunk 1]
```diff
@@ -991,27 +992,36 @@ cdef class WritingSession(Session):
             # TODO: revisit the logic in the following blocks when we
             # change the assumption above.
             else:
-                if collection.driver == "GeoJSON" and os.path.exists(path):
-                    # manually remove geojson file as GDAL doesn't do this for us
+                log.debug("Path exists: path=%r, driver=%r", path, collection.driver)
+
```
Comment (sgillies) [inline fiona/ogrext.pyx:996]:
```suggestion


```

Comment (sgillies) [inline fiona/ogrext.pyx:996]:
```suggestion
```

**Comment by rbuffat** [comment]
@sgillies 
Just FYI / inspiration: In https://github.com/Toblerity/Fiona/pull/909 some parts of the dataset creation are modified.  (https://github.com/Toblerity/Fiona/blob/6ee5939f1bcfe64cbedea454f689f8a1649922f6/fiona/ogrext.pyx#L1026-L1064 (all *_kwargs can be ignored))

Currently, there is a special treatment for the GeoJSON that overwrites local files.  Would it make sense to have an overwrite=True/False flag in fiona.open() instead, that depending of the result of CPLCheckForFile either returns an error or overwrites? If you are interested, I could create a new PR with just this part.

**Comment by sgillies** [comment]
@rbuffat I'm :-1: on an overwrite option. I think the mode argument should cover everything. That said, I think I might be implementing "w" wrong for multi-layer formats. It doesn't truncate to zero (overwrite) in that case, where maybe it should. We could require "r+" to add layers to a multi-layer format?

**Comment by rbuffat** [comment]
> It doesn't truncate to zero (overwrite) in that case, where maybe it should

If I understand correctly the issue is that is not possible to differentiate between creating a new layer, or overwriting an existing file.  If I understand correctly the OGR model strictly separates read and write mode, thus "r+" might be a bit confusing.

Regarding the overwrite aspect, would you just use the existing behavior or having something to separate "w" for writing if the file or layer does not exist and "w!" or "w+" to overwrite a file or layer if it already exists.

**Pull request merged**